src/validation/factory.ts

```diff
@@ -0,0 +1,54 @@
+import { Factory } from 'src/common'
+import { CustomMessages, Severity, Validation, ValidationResult } from './validation'
+
+type ValidationFactoryParams = {
+  errors: Array<ValidationResult>
+  fields: {
+    [name: string]: Validation
+  }
+  valid: boolean
+  warnings: Array<ValidationResult>
+}
+
+const defaultProps = {
+  valid: true,
+  errors: <ValidationResult[]>[],
+  warnings: <ValidationResult[]>[],
```

Comment:
aren't these ```Array<ValidationResult>``` ? not an array of array?

Reply:
yes, we can use `new Array<ValidationResult>()` to make it more readable, as you suggested